helpers/_cookies.tsx

```diff
@@ -1,5 +1,44 @@
+import Cookies from 'js-cookie'
+
 export const COOKIE_NAME_LANGUAGE = 'efw_language'
 export const COOKIE_NAME_ADULT_FILTER = 'efw_adult_filter'
 export const COOKIE_NAME_NEW_TAB = 'efw_new_tab'
 export const COOKIE_NAME_COOKIE_CONSENT = 'efw_cookie_consent_accepted'
 export const COOKIE_NAME_SEARCH_COUNT = 'efw_search_count'
+
+type CookieName =
+  | typeof COOKIE_NAME_LANGUAGE
+  | typeof COOKIE_NAME_ADULT_FILTER
+  | typeof COOKIE_NAME_NEW_TAB
+  | typeof COOKIE_NAME_COOKIE_CONSENT
+  | typeof COOKIE_NAME_SEARCH_COUNT
+
+export type CookieMap = { [cookieName: string]: string }
+
+function convertCookieValue(name: CookieName, value?: string): string | number | boolean {
+  if (value === undefined) {
+    return value
+  }
+
+  switch (name) {
+    case COOKIE_NAME_SEARCH_COUNT:
+    case COOKIE_NAME_LANGUAGE:
+    case COOKIE_NAME_ADULT_FILTER:
+      return Number(value)
+
+    case COOKIE_NAME_NEW_TAB:
+      return value !== 'false'
+
+    default:
+      return value
+  }
+}
+
+export function getCookie(name: CookieName, cookieMap?: CookieMap) {
+  const value = cookieMap ? cookieMap[name] : Cookies.get(name)
+  return convertCookieValue(name, value)
+}
+
+export function setCookie(name: CookieName, value: string, opts?: { expires: number }): void {
```

Comment:
`opts?: { expires: number } ` would this still work if we would put another option? like another property of the opts obj?

Reply:
The idea here is that we'd then extend the opts type. I was trying to prevent leaking the `js-cookie` library's details here which is a fancy way to say: If we want to change the cookies libary later, we'd have to change the interface for `setCookies` in our helper as well which should avoid. So I suggest we only allow the options we really need.

Reply:
:+1: